### PR TITLE
[FIX] Adding Custom Fields to show on user info check

### PR DESCRIPTION
**Pull request by yash-rajpal**
- [x] I have read the Contributing Guide 
  - [x] I have signed the CLA
  - [x] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  - [ ] I have added necessary documentation (if applicable)
  - [ ] Any dependent changes have been merged and published in downstream modules

## Proposed changes (including videos or screenshots)

The setting custom fields to show under user info was not being used when rendering fields in user info. This pr adds those checks and only renders the fields mentioned under in admin -> accounts -> Custom Fields to Show in User Info.

## Issue(s)

Custom Fields to Show in User Info was not being used
#18953 
## Further comments

fix-custom-fields-user-info

### CODE REVIEW

**Comment by yash-rajpal** [comment]
@dougfabris Please review. :)

Added try catch block in case user enters wrong json, json.parse can return error.

**Comment by yash-rajpal** [comment]
@ggazzo Please review.

File: client/views/room/contextualBar/UserInfo/index.js [hunk 1]
```diff
@@ -115,10 +123,10 @@ export const UserInfo = React.memo(function UserInfo({
 				</Info>
 			</>}
 
-			{ customFields && Object.entries(customFields).map(([label, value]) => <React.Fragment key={label}>
+			{ customFields && Object.entries(customFields).map(([label, value]) => (customFieldsToShow.includes(label) ? <> <React.Fragment key={label}>
```
Comment (ggazzo) [inline client/views/room/contextualBar/UserInfo/index.js:126]:
```suggestion
			{ customFields && Object.entries(customFields).filter(([label]) => customFieldsToShow.includes(label)).map(([label, value]) => ( <React.Fragment key={label}>
```

Comment (ggazzo) [inline client/views/room/contextualBar/UserInfo/index.js:126]:
```suggestion
			{ customFields && Object.entries(customFields).filter(([label]) => customFieldsToShow.includes(label)).map(([label, value]) => <React.Fragment key={label}>
```

File: client/views/room/contextualBar/UserInfo/index.js [hunk 2]
```diff
@@ -115,10 +123,10 @@ export const UserInfo = React.memo(function UserInfo({
...
 				<Label>{t(label)}</Label>
 				<Info>{value}</Info>
-			</React.Fragment>) }
+			</React.Fragment> </> : null)) }
```
Comment (ggazzo) [inline client/views/room/contextualBar/UserInfo/index.js:129]:
```suggestion
			</React.Fragment>) }
```

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: client/views/room/contextualBar/UserInfo/UserInfo.js [hunk 3]
```diff
@@ -34,6 +35,18 @@ function UserInfo({
 
 	const timeAgo = useTimeAgo();
 
+	const customFieldsToShowSetting = useSetting('Accounts_CustomFieldsToShowInUserInfo');
+	const customFieldsToShow = [];
+	try {
```
Comment (dougfabris) [inline client/views/room/contextualBar/UserInfo/UserInfo.js:40]: @yash-rajpal we should replace this `try-catch` and use an if statement

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: client/views/room/contextualBar/UserInfo/UserInfo.js (hunk 3 above)
Comment (yash-rajpal) [reply client/views/room/contextualBar/UserInfo/UserInfo.js:40]: Thanks for the review. Done the required Changes. :)
This is my config of custom fields -
image
custom fields to show is 
image

I have also included a Check that if value of any custom field is missing, then we should not render that field, as that would be empty.
image
Label will be rendered as given in json, so showing `roleTest` label inplace of `role`.

image
Here `Test` custom field is missing so not rendering that.

Comment (dougfabris) [reply client/views/room/contextualBar/UserInfo/UserInfo.js:40]: @yash-rajpal I'll check it

File: client/views/room/contextualBar/UserInfo/UserInfo.js [hunk 4]
```diff
@@ -34,6 +35,18 @@ function UserInfo({
...
+	const customFieldsToShow = [];
+	const customFieldsToShowObj = JSON.parse(customFieldsToShowSetting);
+	customFieldsToShowObj &&
+		Object.values(customFieldsToShowObj).map((value) => {
+			const role = Object.values(value);
+			const roleNameToShow = Object.keys(value);
+			const customField = {};
+			customField[roleNameToShow] = customFields[role];
+			return customFieldsToShow.push(customField);
+		});
```
Comment (tassoevan) [inline client/views/room/contextualBar/UserInfo/UserInfo.js:48]: Remember that the `Array.prototype.map` method is not a callback-based for-each loop, but a functional concept:

```suggestion
	const customFieldsToShow = customFieldsToShowObj ?
		Object.values(customFieldsToShowObj).map((value) => {
			const role = Object.values(value);
			const roleNameToShow = Object.keys(value);
			const customField = {};
			customField[roleNameToShow] = customFields[role];
			return customField;
		}) : [];
```

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: client/views/room/contextualBar/UserInfo/UserInfo.js (hunk 4 above)
Comment (yash-rajpal) [reply client/views/room/contextualBar/UserInfo/UserInfo.js]: Thanks @tassoevan for the suggestion. I will remember this. :)

**Review by tassoevan — APPROVED** [review]

**Pull request merged**